pkg/providers/instancetype/offering/provider.go

```diff
@@ -192,22 +207,28 @@ func (p *DefaultProvider) createOfferings(
 	return offerings
 }
 
-func (p *DefaultProvider) cacheKeyFromInstanceType(it *cloudprovider.InstanceType) string {
-	zoneHash, _ := hashstructure.Hash(
+func (p *DefaultProvider) cacheKeyFromInstanceType(it *cloudprovider.InstanceType, subnetZones map[string]string) string {
+	zonesHash, _ := hashstructure.Hash(
```

Comment:
We discussed offline: We should be able to collapse this `zonesHash` and `subnetsZonesHash` into one thing if the InstanceType requirements just only capture the zones and zone ids that this NodeClass specifically selects on